cmd/dra-example-kubeletplugin/state.go

```diff
@@ -151,17 +160,63 @@ func (s *DeviceState) prepareDevices(claim *resourceapi.ResourceClaim) (Prepared
 		return nil, fmt.Errorf("claim not yet allocated")
 	}
 
+	// Walk through each device allocation and prepare it.
 	var preparedDevices PreparedDevices
 	for _, result := range claim.Status.Allocation.Devices.Results {
 		if _, exists := s.allocatable[result.Device]; !exists {
 			return nil, fmt.Errorf("requested GPU is not allocatable: %v", result.Device)
 		}
 
-		device := &drapbv1.Device{
-			RequestNames: []string{result.Request},
-			PoolName:     result.Pool,
-			DeviceName:   result.Device,
-			CDIDeviceIDs: s.cdi.GetClaimDevices([]string{result.Device}),
+		// Retrieve the set of device configs for the current allocation.
+		configs, err := GetOpaqueDeviceConfigs(
+			configapi.Decoder,
+			DriverName,
+			result.Request,
+			claim.Status.Allocation.Devices.Config,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("error getting GPU configs: %v", err)
+		}
+
+		// Select the GPU config with the highest precedence from those retrieved.
+		// Use the default GPU config if no GPU configs have been returned.
+		config := configapi.DefaultGpuConfig()
+		for _, c := range configs {
+			switch castConfig := c.(type) {
+			case *configapi.GpuConfig:
+				config = castConfig
+			default:
+				return nil, fmt.Errorf("runtime object is not a regognized configuration")
```

Comment:
One could also call this an "internal error" because why did the driver register a type that it doesn't recognize here?

But let's leave it as-is.

/lgtm
/approve
/hold

In case you want to get a second opinion (@byako ?), otherwise feel free to lift the hold.

Reply:
Evan did an initial review and had approved it previously. Let's unhold it and I will make any changes as a follow-up if required.